# 중간고사 대체 실습과제 — 문제 1

## 기획의 이해와 6단계 프로세스

| 항목 | 내용 |
|------|------|
| **관련 챕터** | Ch01. 기획이란 무엇인가? |
| **핵심 개념** | 기획의 정의, 분석/계획/전략 구분, 6단계 기획 프로세스 |
| **배점** | 20점 |
| **제출** | 이 노트북(.ipynb)을 실행 결과 포함하여 제출 |

---

## 시나리오

당신은 건강 도시락 구독 서비스 **"프레시밀"**의 기획팀 신입사원입니다.
최근 3개월간 구독 해지율이 15%에서 28%로 급증하고 있습니다.

| 항목 | 수치 |
|------|------|
| 현재 구독자 수 | 3,200명 |
| 월 구독료 | 5만 원 |
| 최근 해지율 | 28% (기존 15%) |
| 월 신규가입 | 120명 |
| 고객 불만 Top 3 | ①배송 지연(35%) ②메뉴 단조(28%) ③가격(22%) |

In [1]:
# 환경설정
import numpy as np
import pandas as pd
import plotly.graph_objects as go
from plotly.subplots import make_subplots

In [2]:
# 데이터 로드
monthly = pd.DataFrame({
    '월': ['10월', '11월', '12월', '1월', '2월', '3월'],
    '구독자수': [3800, 3650, 3500, 3400, 3300, 3200],
    '해지율_pct': [15, 18, 22, 25, 27, 28],
    '배송지연건수': [45, 62, 88, 110, 125, 140],
    '고객만족도': [4.2, 4.0, 3.7, 3.4, 3.2, 3.0]
})

complaint = pd.DataFrame({
    '불만유형': ['배송 지연', '메뉴 단조', '가격 불만', '포장 품질', '기타'],
    '비율_pct': [35, 28, 22, 10, 5]
})

print(monthly.to_string(index=False))
print()
print(complaint.to_string(index=False))

  월  구독자수  해지율_pct  배송지연건수  고객만족도
10월  3800       15      45    4.2
11월  3650       18      62    4.0
12월  3500       22      88    3.7
 1월  3400       25     110    3.4
 2월  3300       27     125    3.2
 3월  3200       28     140    3.0

 불만유형  비율_pct
배송 지연      35
메뉴 단조      28
가격 불만      22
포장 품질      10
   기타       5


---

## 과제 1. 개념 이해 (6점)

프레시밀 사례를 활용하여 답하세요.

### 요구사항

1. **(3점)** "분석", "기획", "전략"의 차이를 프레시밀 사례에 적용하여 **각각 1문장**으로 설명하세요.

2. **(3점)** 프레시밀의 해지율 대응은 "전략기획", "정책기획", "사업기획" 중 어디에 해당하는지 고르고 이유를 서술하세요.

### ✏️ 과제 1 답안

**1. 분석/기획/전략 구분**
- 분석: 프레시밀 해지 증가 원인을 분석해서 배송 지연, 메뉴 단조 등의 주요 원인을 찾는 과정이다.
- 기획: 분석한 데이터를 기반으로 배송 지연 해결, 메뉴 다양화 등 해지율을 줄이기 위한 방안을 설계하는 과정이다.
- 전략: 경쟁 대상보다 우위에 있기 위해 배송 안정성과 대상과의 식단 차별화를 중심으로 장기적인 목표를 정한다.

**2. 기획 유형**

- 유형: (전략기획)
- 이유: (프레시밀의 해지율 대응은 전략기획이다.프레시밀의 해지율에 대한 대응은 그들의 운영 방식을 개선하여 배송 지연과 메뉴 단조라는 문제점을 배송 안정성과 다른 기업과의 식단 차별화를 둠으로써 장기적인 방향성과 경쟁에서 우위를 확보하기 위한 과정이라고 볼 수 있다. 따라서 전략기획에 해당한다.)

---

## 과제 2. 6단계 기획 프로세스 적용 (8점)

```
1. 문제 정의  →  2. 원인 분석  →  3. 대안 개발
→  4. 의사결정  →  5. 실행  →  6. 학습
```

### 요구사항

1. **(3점)** 제공된 데이터를 시각화하세요:
   - (a) 월별 해지율 추이 라인 차트
   - (b) 불만 유형 비율 파이 차트

2. **(5점)** 6단계 프로세스 중 **1~4단계**를 아래 양식에 작성하세요:
   - 1단계(문제 정의): 핵심 문제를 1~2문장으로
   - 2단계(원인 분석): 데이터에서 확인되는 원인 2개
   - 3단계(대안 개발): 해결 대안 3개
   - 4단계(의사결정): 가장 효과적인 대안 1개 선택 + 이유

In [3]:
# ✏️ 과제 2-1: 시각화
# (a) 월별 해지율 라인 차트
fig_line = go.Figure()
fig_line.add_trace(go.Scatter(
    x=monthly['월'],
    y=monthly['해지율_pct'],
    mode='lines+markers',
    name='해지율(%)',
    line=dict(color='black', width=2),
    marker=dict(size=8, color='white', line=dict(color='black', width=2))
))

fig_line.update_layout(
    title='월별 구독 해지율 추이',
    xaxis_title='월',
    yaxis_title='해지율 (%)',
    template='plotly_white',
    font=dict(color='black'),
    hovermode='x unified'
)
fig_line.show()

# (b) 불만 유형 비율 파이 차트
fig_pie = go.Figure(data=[go.Pie(
    labels=complaint['불만유형'],
    values=complaint['비율_pct'],
    marker=dict(line=dict(color='black', width=2)),
    textfont=dict(color='black')
)])

fig_pie.update_layout(
    title='고객 불만 유형별 비율',
    font=dict(color='black')
)
fig_pie.show()

print("✓ 시각화 완료: 해지율 추이와 불만 유형 분포를 확인했습니다.")

✓ 시각화 완료: 해지율 추이와 불만 유형 분포를 확인했습니다.


### ✏️ 과제 2 해석 (이 셀에 직접 타이핑)

시각화 결과를 보고 아래에 답하세요.

- 해지율 추이에서 가장 눈에 띄는 변화 시점은? 그 이유는 무엇이라 추측하는가?
해지율 추이에서 12월부터 가장 눈에 띈다고 생각한다. 왜냐하면 12월부터 해지율 추이가 20%를 넘었고 그 이후로 부터 25%,27%,28%로 점점 늘어나고 있다. 따라서 12월 시점 부터 배송 지연과 메뉴 단조라는 단점이 부각되었다고 생각한다.
- 불만 유형 중 가장 비율이 높은 항목이 해지율과 어떻게 연결되는가? (1~2문장)
배송지연이 35%로 가장 높은 비율을 차지하고 있다. 배송지연으로 인해 손님들의 인내심에 한계를 느꼈을 것이고 신뢰감이 떨어져 다른 경쟁 업체를 구독했을 것이다.

### ✏️ 과제 2-2: 6단계 프로세스 (1~4단계)

**1단계. 문제 정의:** (프레시밀의 월별 구독 해지율이 갈수록 10%대에서 거의 30%대에 가까운 수치로 급증하였다. 이는 고객들이 프레시밀에 대한 불만은 갈수록 높아지고 만족도는 떨어진다고 볼 수 있다.)

**2단계. 원인 분석:**
- 원인 1: (가장 큰 원인은 배송지연으로 총 35%를 차지하고 있다. 배송지연 건수가 45건에서 140건으로 늘어났다.)
- 원인 2: (메뉴 다양성이 부족하고 가격 문제가 절반 정도의 비율을 차지하고 있다.)

**3단계. 대안 개발:**

| 대안 | 내용 |
|------|------|
| 대안 1 |배송 지연 해결: 배송 출발지를 세분화해서 최적화로 배송을 해서 지연을 감소시킨다. |
| 대안 2 |메뉴 단조 해결: 메뉴를 다양하게 늘리고 설문조사를 받아 손님들이 선호하는 맛의 메뉴를 출시한다. |
| 대안 3 |가격 문제 해결: 오픈기념일 이벤트나 할인 이벤트, 장기 구독자 할인, 다른 기관과의 콜라보 등으로 손님들에게 더 좋은 가격과 이벤트를 열어 만족도를 높인다.  |

**4단계. 의사결정:** (배송 지연을 우선적으로 해결해야하므로 대안 1을 최우선으로 선택해야한다. 그래프로 봐도 배송지연에 대한 불만감이 가장 높고 지연건수가 100개 정도의 수량이 더 늘어났기 때문에 이대로 유지시키면 지연 건수는 더 늘어나고 불만감은 더 높아질 것이다.)

---

## 과제 3. 간단한 시뮬레이션 (6점)

현재 상태(구독자 3,200명, 해지율 28%, 신규 120명/월)로 **6개월 후 구독자 수**를 계산하세요.

### 요구사항

1. **(3점)** 아래 두 시나리오의 6개월간 구독자 수 변화를 계산하세요:
   - 시나리오 A: 현상 유지 (해지율 28%)
   - 시나리오 B: 배송 개선으로 해지율 20%로 감소

2. **(3점)** 두 시나리오의 구독자 수 추이를 **라인 차트**로 비교하고, 6개월 후 차이를 `print()`로 출력하세요.

In [5]:
# ✏️ 과제 3 풀이
# 시나리오 A vs B 시뮬레이션 + 라인 차트 + 결론 print()

months = [1, 2, 3, 4, 5, 6]
start_customers = 3200
new_customers = 120

# 계산 함수

def simulate_customers(start, churn_rate, new_monthly, months):
    customers = start
    history = []
    for _ in months:
        churn = customers * churn_rate
        customers = customers - churn + new_monthly
        history.append(customers)
    return history

scenario_a = simulate_customers(start_customers, 0.28, new_customers, months)
scenario_b = simulate_customers(start_customers, 0.20, new_customers, months)

fig_sim = go.Figure()
fig_sim.add_trace(go.Scatter(
    x=months,
    y=scenario_a,
    mode='lines+markers',
    name='시나리오 A: 현상 유지 (28%)',
    line=dict(color='black', width=2),
    marker=dict(size=8, color='white', line=dict(color='black', width=2))
))
fig_sim.add_trace(go.Scatter(
    x=months,
    y=scenario_b,
    mode='lines+markers',
    name='시나리오 B: 배송 개선 (20%)',
    line=dict(color='gray', width=2, dash='dash'),
    marker=dict(size=8, color='white', line=dict(color='black', width=2))
))

fig_sim.update_layout(
    title='6개월 후 구독자 수 시나리오 비교',
    xaxis_title='개월',
    yaxis_title='구독자 수',
    template='plotly_white',
    font=dict(color='black'),
    hovermode='x unified'
)
fig_sim.show()

final_a = scenario_a[-1]
final_b = scenario_b[-1]
difference = final_b - final_a

print(f"시나리오 A 6개월 후 구독자 수: {int(final_a)}명")
print(f"시나리오 B 6개월 후 구독자 수: {int(final_b)}명")
print(f"6개월 후 시나리오 B가 시나리오 A보다 {int(difference)}명 더 유지됩니다.")

시나리오 A 6개월 후 구독자 수: 814명
시나리오 B 6개월 후 구독자 수: 1281명
6개월 후 시나리오 B가 시나리오 A보다 466명 더 유지됩니다.


### ✏️ 과제 3 해석 (이 셀에 직접 타이핑)

- 시나리오 A와 B의 6개월 후 구독자 수 차이는 몇 명인가?
시나리오 B가 A보다 466명 더 많다.
- 이 차이가 **매출(월 구독료 5만 원)**로 환산하면 얼마인가?
23,300,000원
- 이 결과가 경영진에게 시사하는 바를 1문장으로 작성하세요.
배송지연을 개선해서 운영하면 월 23,300,000원의 매출을 추가로 확보할 수 있으므로 배송 개선은 필수적으로 해야한다.

---

## 채점 기준

| 과제 | 배점 | 채점 포인트 |
|------|------|------------|
| 과제 1. 개념 이해 | 6점 | 분석/기획/전략 구분(3) + 기획유형(3) |
| 과제 2. 6단계 프로세스 | 8점 | 시각화(3) + 프로세스 적용(5) |
| 과제 3. 시뮬레이션 | 6점 | 계산(3) + 시각화+결론(3) |
| **합계** | **20점** | |

---

## 참고: 예상 정답값

**과제 1 예시:**
- 분석: "과거 6개월 해지 데이터를 분석하여 배송 지연이 주 원인임을 파악하는 것"
- 기획: "해지율을 15%로 되돌리기 위한 구체적 행동 계획을 설계하는 것"
- 전략: "프리미엄 건강식 시장의 1위라는 방향성을 설정하는 것"

**과제 3 시뮬레이션 (6개월 후 근사치):**

| 시나리오 | 6개월 후 구독자 수 |
|---------|-------------------|
| A. 현상 유지 (28%) | ~816명 |
| B. 배송 개선 (20%) | ~1,283명 |

- 배송 개선 시 약 467명 더 유지 가능